pkgs/build-support/rust/default.nix

```diff
@@ -94,7 +94,9 @@ in stdenv.mkDerivation (args // {
   installPhase = args.installPhase or ''
     runHook preInstall
     mkdir -p $out/bin
-    find target/release -maxdepth 1 -executable -type f -exec cp "{}" $out/bin \;
+    mkdir -p $out/lib
+    find target/release -maxdepth 1 -type f -executable ! \( -name "*.so.*" -o -name "*.so" -o -name "*.a" -o -name "*.dylib" \) | xargs -r -n 1 cp -t $out/bin
```

Comment:
Can you switch to regex here? In particular we should only allow `[0-9.]` after `.so`. Also `-print0` for find and `xargs -0` is considered safer.

Reply:
Does `cp` not allow multiple arguments to be passed?

Reply:
Switched to regex in latest commit in order to match [0-9] instead of any arbitrary character and -print0 with xargs -0. `cp` allows multiple arguments to be passed, if you can find a better instantiation of the command which works then will change.